### ISSUE TITLE
.with_options() prevents pulling config from top-level

### ISSUE THREAD

**Issue by willkg** [willkg/everett#16]
Say we had a `DEBUG` configuration option that's defined for the entire application. The problem with `.with_options()` is that it constrains the configuration for a component to a specific namespace preventing it from looking at other parts of the configuration. Thus there's no way for a component to look at an app-scoped `DEBUG` configuration option.

This issue covers mulling over that because that kind of sucks for a large number of use cases: debug flags, logging, etc.

Maybe the solution is for the app to pull these configuration and pass it along in the `__init__`?

**Comment by willkg** [comment]
I just implemented something that's like this scenario. What I ended up doing was having the app pull out the configuration item (in this case, basedir) and then pass it along to the component alongside the configuration.

e.g.

```
class Component(RequiredConfigMixin):
    ...
    def __init__(self, config, basedir):
        self.config = config.with_options(self)
        self.basedir = basedir
```

I think there are a few ways to talk about this, but I think the way I want to think about it is that "basedir" is not configuration for this component and that it's external configuration and affected by rules that are external to this component. I claim it's more flexible for this component to not have to know where in the configuration hierarchy to pull "basedir" from especially where it's defined who-knows-where.

Given that, I think at this stage I don't want to make any changes to Everett, but instead want to document this scenario as a recipe.
